modules/monitoring-resolving-the-kubepersistentvolumefillingup-alarm-firing-for-prometheus.adoc

```diff
@@ -0,0 +1,67 @@
+// Module included in the following assemblies:
+//
+// * monitoring/troubleshooting-monitoring-issues.adoc
+// * support/troubleshooting/investigating-monitoring-issues.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="resolving-the-kubepersistentvolumefillingup-alarm-firing-for-prometheus_{context}"]
+= Resolving the KubePersistentVolumeFillingUp alarm firing for Prometheus
+
+As a cluster administrator, you can resolve the `KubePersistentVolumeFillingUp` alarm being triggered for Prometheus. 
+
+The alert fires when a persistent volume (PV) claimed by a `prometheus-k8s-*` pod in the `openshift-monitoring` project has less than 3% of its total space remaining. This can cause Prometheus to not function normally.
```

Comment:
The [KCS article](https://access.redhat.com/solutions/7033177) mentions these issues:
* **The PersistentVolume claimed by prometheus-k8s-db-prometheus-k8s-X in Namespace openshift-monitoring is only 0% free.**
* **Prometheus pod prometheus-k8s-db-prometheus-k8s-X in CrashLoopBackOff state.**

Are these the alert messages the user would get from the `KubePersistentVolumeFillingUp` alarm?

Reply:
> The [KCS article](https://access.redhat.com/solutions/7033177) mentions these issues:
> 
>     * **The PersistentVolume claimed by prometheus-k8s-db-prometheus-k8s-X in Namespace openshift-monitoring is only 0% free.**

see from https://github.com/openshift/openshift-docs/pull/73686#discussion_r1540618820, the message is from KubePersistentVolumeFillingUp alert, the variables are replaced by actual value

>     * **Prometheus pod prometheus-k8s-db-prometheus-k8s-X in CrashLoopBackOff state.**
this should be from `KubePodCrashLooping` alert, if there is `KubePersistentVolumeFillingUp` alert fired, I think it will trigger `KubePodCrashLooping`alert after some time. wait for others to confirm
```
$ oc -n openshift-monitoring get prometheusrules kubernetes-monitoring-rules -oyaml
```
    - alert: KubePodCrashLooping
      annotations:
        description: 'Pod {{ $labels.namespace }}/{{ $labels.pod }} ({{ $labels.container
          }}) is in waiting state (reason: "CrashLoopBackOff").'
        summary: Pod is crash looping.
      expr: |
        max_over_time(kube_pod_container_status_waiting_reason{reason="CrashLoopBackOff", namespace=~"(openshift-.*|kube-.*|default)",job="kube-state-metrics"}[5m]) >= 1
      for: 15m
      labels:
        severity: warning
```

Reply:
I see, thank you for the confirmation! If that is the case, then I think it would be enough to keep the original sentence as it is:

`The alert fires when a persistent volume (PV) claimed by a prometheus-k8s-* pod in the openshift-monitoring project has less than 3% of its total space remaining. This can cause Prometheus to not function normally.`

as it already mentions possible problems with functionality. But I will wait for the confirmation. Thank you!

Reply:
I'm fine with 

> The alert fires when a persistent volume (PV) claimed by a prometheus-k8s-* pod in the openshift-monitoring project has less than 3% of its total space remaining. This can cause Prometheus to not function normally.

Lack of disk space can trigger many different problems and we can list/anticipate them all